src/lib.rs

```diff
@@ -81,6 +80,7 @@ impl<E> From<E> for AsyncMutexError<E> {
 enum AcquireFutureState<T, F, G> {
     WaitResource((oneshot::Receiver<T>, F)),
     WaitFunction(G),
+    Broken,
```

Comment:
@juchiast : Why did we add a Broken state here? Sorry that I noticed it only now, I think that it was added in one of the earlier commits.
Does the original code of AcquireFutureState could not work with the new design of ResourceState?

Reply:
@realcr Previously, `acquire()` a broken resource will resolve to `WaitResource`, then it will check `is_broken` in `poll()`. Now with this change, it will return `Broken` immediately, instead of `WaitResource`. I added it because returning `WaitResource` would be a little ambiguous.